src/fsharp/ConstraintSolver.fs

```diff
@@ -2136,31 +2136,39 @@ and ResolveOverloading
           // See what candidates we have based on current inferred type information 
           // and _exact_ matches of argument types. 
           match candidates |> FilterEachThenUndo (fun newTrace calledMeth -> 
+                match cx with 
+                | None -> ResultD true
+                | Some traitInfo -> RecordMemberConstraintSolution csenv.SolverState m (WithTrace newTrace) traitInfo (TTraitSolved (calledMeth.Method, calledMeth.CalledTyArgs))                
+                ++ (fun _ ->
                      CanMemberSigsMatchUpToCheck 
                          csenv 
                          permitOptArgs 
                          alwaysCheckReturn
                          (MustUnifyInsideUndo csenv ndeep newTrace) 
                          (TypesMustSubsumeOrConvertInsideUndo csenv ndeep (WithTrace newTrace) m)
-                         (ArgsEquivInsideUndo csenv Trace.New isConstraint) 
+                         (ArgsEquivInsideUndo csenv Trace.New cx.IsSome) 
```

Comment:
Good point @dsyme, looks like a bug. Because if a new trace is used then <code>FilterEachThenUndo</code> will not undo those changes.
Now if this is a bug that has no consequences I wonder if that operation must be inside an undo at all.